addons/package_hierarchy/models/package_links.py

```diff
@@ -52,6 +52,57 @@ class PackageHierarchyLink(models.Model):
     )
     company_id = fields.Many2one("res.company", default=lambda self: self.env.company)
 
+    @api.model
+    def create(self, vals):
+        """
+        Extend create to prevent creating duplicate parent/child relations
+        and instead return the duplicate. This is in place of a constraint
+        as it is more user friendly
+        (instead of raising an error will simply return the existing link)
+        """
+        PackageHierarchyLink = self.env["package.hierarchy.link"]
```

Comment:
nit: I think there should be vertical space after defining the model?

Reply:
Sure, i'll get this sorted